cmd/file-logger.go

```diff
@@ -0,0 +1,96 @@
+/*
+ * Minio Cloud Storage, (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"os"
+
+	"github.com/Sirupsen/logrus"
+)
+
+// FileLogger - file logger which logs to a file.
+type FileLogger struct {
+	BaseLogTarget
+	Filename string `json:"fileName"`
+	file     *os.File
+}
+
+// Fire - log entry handler.
+func (logger FileLogger) Fire(entry *logrus.Entry) (err error) {
+	if !logger.Enable {
+		return nil
+	}
+
+	if logger.Level.IsHigher(entry.Level) {
+		return nil
+	}
+
+	msgBytes, err := logger.formatter.Format(entry)
+	if err != nil {
+		return err
+	}
+
+	if _, err = logger.file.Write(msgBytes); err != nil {
+		return err
+	}
+
+	err = logger.file.Sync()
+	return err
+}
+
+// String - represents ConsoleLogger as string.
+func (logger FileLogger) String() string {
+	str := "file"
+
+	if logger.Enable {
+		str += ":enabled"
+	} else {
+		str += ":disabled"
+	}
+
+	str += ":" + logger.Level.String()
+
+	str += ":" + logger.Filename
+	return str
+}
```

Comment:
This function is better implemented with Sprintf, than by so many concatenations:

```go
enableStr := "enabled"
if !logger.Enable {
       enableStr = "disabled"
}

return fmt.Sprintf("file:%s:%s:%s", enableStr, logger.Level.String(), logger.Filename)
```

Arguably this is more readable.